query-string-parser.js

```diff
@@ -4,20 +4,13 @@ const lodash = require('lodash')
 
 exports.parse = parse
 
-const DefaultDurationSeconds = 5
-
 const Groups = {
-  agentScope: parseAgentScope,
-  duration: parseDuration,
-  nameData: parseNameData,
-  fields: parseFields,
-  interval: parseInterval,
-  startEnd: parseStartEnd,
-  agg: parseAgg
 }
 
-for (let key of ['id', 'app', 'key', 'val', 'name', 'data', 'asset', 'type', 'firstName', 'lastName', 'email', 'company', 'optOut', 'license']) {
-  Groups[key] = getSelfParser(key)
+exports.register = function (params) {
+  if (params.key && !Groups[params.key]) {
+    Groups[params.key] = params.validator || getSelfParser(params.key)
```

Comment:
Uggh; I feel like this module wasn't really designed well to begin with, seems like it's not surviving a refactor well!  Heh.  Example is the names here, `Groups` doesn't make much sense in this case where it's really keys (or so), and `getSelfParser()` - what does it mean???  OMG.

So, I think this module deserves some major cleanup, but **it's not critica**l.  One consideration is that since apimule is a lib instead of part of a specific app, prolly want to make this module less global, in case someone wants to have different sets of query parms, for different servers, in the same app.

Just opened issue https://github.com/nodesource/apimule/issues/5 to track that.